sklearn/preprocessing/imputation.py

```diff
@@ -143,27 +147,35 @@ def fit(self, X, y=None):
                              " got strategy={1}".format(allowed_strategies,
                                                         self.strategy))
 
-        if self.axis not in [0, 1]:
+        if self.axis is None:
+            self.axis_ = 0
```

Comment:
This attribute implicitly becomes part of the public API. Either use a private attribute (beginning _) or use a local variable

Reply:
Ups my bad ... I should have advise to make it private